src/solvers/smt2_incremental/convert_expr_to_smt.cpp

```diff
@@ -1164,6 +1164,29 @@ static smt_termt convert_expr_to_smt(
   return extract_op;
 }
 
+static smt_termt convert_expr_to_smt(
+  const pointer_offset_exprt &pointer_offset,
+  const sub_expression_mapt &converted)
+{
+  const auto type =
+    type_try_dynamic_cast<bitvector_typet>(pointer_offset.type());
+  INVARIANT(type, "Pointer offset should have a bitvector-based type.");
+  const auto pointer_expr = pointer_offset.pointer();
+  const auto converted_expr = converted.at(pointer_expr);
+  const std::size_t pointer_width = type->get_width();
+  std::size_t offset_bits = pointer_width - config.bv_encoding.object_bits;
+  if(offset_bits > pointer_width)
+    offset_bits = pointer_width;
+  const auto extract_op =
+    smt_bit_vector_theoryt::extract(offset_bits - 1, 0)(converted_expr);
+  if(pointer_width > offset_bits)
+  {
+    return smt_bit_vector_theoryt::zero_extend(pointer_width - offset_bits)(
```

Comment:
:thinking: The result type of `pointer_offset_exprt` is usually signed. Therefore we should probably be using `sign_extend` instead of `zero_extend`. Note that the old SMT2 back end may may wrong in this case. Note also that the signed-ness is currently set in `pointer_predicates.cpp` but we have no guarantees that these expressions will always have signed/unsigned results (unless we add appropriate invariants).
:information_source: Negative offsets may occur in the the case of pointer offset overflows.